Tor.0.4.5.7 using openssl 1.0.x

svschmel opened the pull request:
Hello,

with PR #6586 i wanted to merge the new tor version 0.4.5.7 into OI.
That PR was built using openssl 1.1 which is not the default on OI at this moment.

This new PR is using openssl 1.0.
So we can merge this new tor version in OI.

PR #6586 can be used for the future or Andreas add the openssl-1.1 changes to his big PR.

Info about this PR:

When starting tor a new information about the used libc is posted.
For OI (and other) it is "... and Unknown N/A as libc ..."

This is OK because the function is rudimentary:

```
tor_libc_get_version_str(void)
{
#ifdef CHECK_LIBC_VERSION
  const char *version = gnu_get_libc_version();
  if (version == NULL)
    return "N/A";
  return version;
#else /* !defined(CHECK_LIBC_VERSION) */
  return "N/A";
#endif /* defined(CHECK_LIBC_VERSION) */
```

"gmake test" has one error:

```
PASS: src/test/unittest_part7.sh
FAIL: src/test/unittest_part8.sh
PASS: src/test/test_ntor.sh
PASS: src/test/test_hs_ntor.sh
PASS: src/test/test_bt.sh
PASS: scripts/maint/practracker/test_practracker.sh
PASS: scripts/maint/run_check_subsystem_order.sh
PASS: src/test/test_rebind.sh
PASS: src/test/test_include.sh
PASS: scripts/maint/checkSpaceTest.sh
# TOTAL: 33
# PASS:  30
# SKIP:  2
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
```
The error in "src/test/unittest_part8.sh" is part of issue:
https://gitlab.torproject.org/tpo/core/tor/-/issues/40335

All work fine.
https://metrics.torproject.org/rs.html#details/AD0FCFD83EA4899DB34735E599701592DEA13690

svschmel committed:
tor: Update to version 0.4.5.7

AndWac commented:
Can you squash the commits into one before I merge the PR, please?

AndWac commented:
Hi, do you need help with squashing the commits?

svschmel commented:
Hello Andreas,
sorry for the late answer (too busy. Outage of one of our internal LDAP Server :( )

Yes, i need help with squashing the commits.
I saw your answer to Gary Mills on 20.03.21 at 18:37h but this does not work for me:

with "git rebase -i HEAD~2" i see:


```
pick c02daeef6 tor: Update to version 0.4.5.7 with openssl 1.0.x
pick 050781e29 goffice: update to 0.10.49
pick 6b9a801e3 iso-codes: update to 4.6.0
pick 3a886a3a3 Update jenkins-core-weekly to 2.284 and jenkins-core-lts to 2.277.1
pick 54256ff71 nvidia-460.67
pick 9bb44e603 fish: update to 3.2.1
pick 6f3bab23c gexiv2: update to 0.12.2
pick 2e9c176ea squeak-4: update 4.19.5 revision 3
pick 70a49702b squeak-5: add display-X11 package
pick ce55c9514 updating golang-116
pick ec6b6f363 updating golang-114
pick 7861f6bc0 updating golang-115
pick 6d6546139 bash: update to 5.1.p4
pick 147934d8c squeak-4: update to 4.19.6
pick 16b12a05e zziplib: update to 0.13.72
....
```
Only my second PR is mentioned. Perhaps i made a mistake.

Greetings
Sven

AndWac commented:
What does "git log" tell? Are you sure you are in the right branch? Is the branch rebased against the actual oi/hipster repository?

svschmel commented:
Hello,
the rebase of the branch was the trick. sorry

```
1.
svschmel@oibuild:~/oi-userland/components/network/tor$ git branch
  help
  libevent-2.1.12
  oi/hipster
  tor.0.4.3.6
  tor.0.4.4.5
  tor.0.4.4.6
  tor.0.4.4.7
  tor.0.4.5.7_1
* tor.0.4.5.7_2
```

```
2.
svschmel@oibuild:~/oi-userland/components/network/tor$ git pull --rebase upstream oi/hipster
remote: Enumerating objects: 65, done.
remote: Counting objects: 100% (65/65), done.
remote: Compressing objects: 100% (20/20), done.
remote: Total 42 (delta 29), reused 27 (delta 19), pack-reused 0
Unpacking objects: 100% (42/42), 7.40 KiB | 15.00 KiB/s, done.
From https://github.com/OpenIndiana/oi-userland
 * branch                oi/hipster -> FETCH_HEAD
   0cc37bd59..cac0b476a  oi/hipster -> upstream/oi/hipster
dropping 3ae8c67200fa73a5d78a9081206b26ed937c5a29 goffice: update to 0.10.49 -- patch contents already upstream
dropping ca72c0651ef8a2e572b90d7008f6878392309a6e iso-codes: update to 4.6.0 -- patch contents already upstream
dropping 6314be66b6fd3475553a26a7eb811ed76ed545db Update jenkins-core-weekly to 2.284 and jenkins-core-lts to 2.277.1 -- patch contents already upstream
Successfully rebased and updated refs/heads/tor.0.4.5.7_2.
```

```
3.
svschmel@oibuild:~/oi-userland/components/network/tor$ git log
commit d8297e8936722c26d18452f7df7e42adf98748fb (HEAD -> tor.0.4.5.7_2)
Author: svschmel <sven.schmeling@schmeling-ol.de>
Date:   Fri Mar 19 08:44:42 2021 +0100

    tor: Update to version 0.4.5.7 with openssl 1.0.x

commit 8752df9c021b8d46ded0c10b09374a3653ada27c
Author: svschmel <sven.schmeling@schmeling-ol.de>
Date:   Thu Mar 18 07:56:31 2021 +0100

    tor: Update to version 0.4.5.7

commit cac0b476a9d72cb145df98f50acd1b02bcc22eb6 (upstream/oi/hipster)
Author: Andreas Wacknitz <A.Wacknitz@gmx.de>
Date:   Fri Mar 26 08:16:43 2021 +0100

    eom: update to 1.24.2

commit 4250c074cce9f563c2960c6efd81df4f2da13f68
Author: Andreas Wacknitz <A.Wacknitz@gmx.de>
Date:   Fri Mar 26 08:16:27 2021 +0100

    atril: update to 1.24.1
....
```

```
4.
git rebase -i HEAD~2
=> 2 Entries
first "pick"
second "squash"

[detached HEAD 55d6521f0] tor: Update to version 0.4.5.7
 Date: Thu Mar 18 07:56:31 2021 +0100
 3 files changed, 22 insertions(+), 16 deletions(-)
Successfully rebased and updated refs/heads/tor.0.4.5.7_2.

```

```
5.
svschmel@oibuild:~/oi-userland/components/network/tor$ git log
commit 55d6521f0e46bbf2e6fdb160b4ca99ef90d58f7d (HEAD -> tor.0.4.5.7_2)
Author: svschmel <sven.schmeling@schmeling-ol.de>
Date:   Thu Mar 18 07:56:31 2021 +0100

    tor: Update to version 0.4.5.7
    
    tor: Update to version 0.4.5.7 with openssl 1.0.x

commit cac0b476a9d72cb145df98f50acd1b02bcc22eb6 (upstream/oi/hipster)
Author: Andreas Wacknitz <A.Wacknitz@gmx.de>
Date:   Fri Mar 26 08:16:43 2021 +0100

    eom: update to 1.24.2

```

```
6.
git push -f origin tor.0.4.5.7_2
Enter passphrase for key '/export/home/svschmel/.ssh/id_rsa': 
Enumerating objects: 19, done.
Counting objects: 100% (19/19), done.
Delta compression using up to 2 threads
Compressing objects: 100% (9/9), done.
Writing objects: 100% (10/10), 1.19 KiB | 1.19 MiB/s, done.
Total 10 (delta 6), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (6/6), completed with 6 local objects.
To ssh://github.com/svschmel/oi-userland
 + 7675e13d1...55d6521f0 tor.0.4.5.7_2 -> tor.0.4.5.7_2 (forced update)

```

Is there something else what i must do?

AndWac commented:
Yes, you should have updated the comments: It's still two lines, one of them being superfluous:

    tor: Update to version 0.4.5.7

    tor: Update to version 0.4.5.7 with openssl 1.0.x

In fact, Github can squash commits automatically but not squash the commit notes so that's the main reason to squash locally ;-)
Now you'll need to
git commit --amend
locally and fix the comments and then force push again.

svschmel commented:
done
